### ISSUE TITLE
[Bug]: internal error: panic runtime error: index out of range [10] with length 10 when running sysbench random point case with 1000w data

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#14509]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.1-dev

### Commit ID

40f1df075c8df5abfabab6bf5da445051d26a5a1

### Other Environment Information

```Markdown
- Hardware parameters: 64C 128G 
- OS type:
- Others:
```

### Actual Behavior

job link:
https://github.com/matrixorigin/mo-nightly-regression/actions/runs/7741895346/job/21110133785

when running sysbench random point case with 1000w data, all the queries were failed.
FATAL: mysql_store_result() returned error 20101 (internal error: panic runtime error: index out of range [10] with length 10: 
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:114
github.com/matrixorigin/matrixone/pkg/sql/plan/function.PrefixIn
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/plan/function/func_prefix.go:135
github.com/matrixorigin/matrixone/pkg/sql/colexec.(*FunctionExpressionExecutor).Eval
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/col)
FATAL: `thread_run' function failed: select_random_points.lua:71: SQL error, errno = 20101, state = 'HY000': internal error: panic runtime error: index out of range [10] with length 10: 
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:114
github.com/matrixorigin/matrixone/pkg/sql/plan/function.PrefixIn
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/plan/function/func_prefix.go:135
github.com/matrixorigin/matrixone/pkg/sql/colexec.(*FunctionExpressionExecutor).Eval
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/col
FATAL: mysql_store_result() returned error 20101 (internal error: panic runtime error: index out of range [10] with length 10: 
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:114
github.com/matrixorigin/matrixone/pkg/sql/plan/function.PrefixIn
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/plan/function/func_prefix.go:135
github.com/matrixorigin/matrixone/pkg/sql/colexec.(*FunctionExpressionExecutor).Eval
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/col)
FATAL: `thread_run' function failed: select_random_points.lua:[71](https://github.com/matrixorigin/mo-nightly-regression/actions/runs/7741895346/job/21110133785#step:13:72): SQL error, errno = 20101, state = 'HY000': internal error: panic runtime error: index out of range [10] with length 10: 
runtime.goPanicIndex
	/usr/local/go/src/runtime/panic.go:114
github.com/matrixorigin/matrixone/pkg/sql/plan/function.PrefixIn
	/data1/runners/action-runner/_work/mo-nightly-regression/mo-nightly-regression/head/pkg/sql/plan/function/func_prefix.go:135
github.com/matrixorigin/matrixone/pkg/sql/colexec.(*FunctionExpressionExecutor).Eval

### Steps to Reproduce

```Markdown
1. git clone https://github.com/matrixorigin/mo-sysbench.git
2. prepare data:
   create database sysbench_1000w_auto_index;
   sysbench --mysql-host=127.0.0.1 --mysql-port=6001 --mysql-user=dump --mysql-password='111' oltp_common.lua --mysql-db=sysbench_1000w_auto_index --tables=10 cleanup
   sysbench --mysql-host=127.0.0.1 --mysql-port=6001 --mysql-user=dump --mysql-password='111' oltp_common.lua --mysql-db=sysbench_1000w_auto_index --tables=10 --table_size=10000000 --threads=10  prepare

3. run test:
   sysbench  --mysql-host=127.0.0.1 --mysql-port=6001 --mysql-user=dump --mysql-password=111 --db-ps-mode=disable select_random_points.lua --mysql-db=sysbench_10w_auto_index --tables=10 --table_size=100000 --threads=100 --time=60 --report-interval=2 run
```

**Comment by aunjgr** [comment]
fixed by #14511 and #14519

**Comment by aressu1985** [comment]
fixed
